bid-distribution/src/settlement_claims.rs

```diff
@@ -38,12 +38,13 @@ pub fn generate_bid_settlements(
     settlement_config: &SettlementConfig,
 ) -> Vec<Settlement> {
     info!("Generating bid settlements...");
-    let sam_validators_epoch = sam_validator_metas.first().unwrap().epoch;
-
-    assert_eq!(
-        stake_meta_index.stake_meta_collection.epoch, sam_validators_epoch as u64,
-        "SAM Validators Collection epoch must be same as stake meta collection epoch"
-    );
+    
+    for sam_validator_meta in sam_validator_metas {
```

Comment:
This is just a super nitpick just thinking how I read the code. When I can see the `for` loop it seems that something should happen, like some mutation.
When there is just an assert with some function it feels for the first grasp being just some assertion, like this:

```
    assert!(
        sam_validator_metas.iter().all(|v| v.epoch as u64 == stake_meta_index.stake_meta_collection.epoch), 
        "SAM Validators Collection epoch must be same as stake meta collection epoch"
    );
    ```

Reply:
you're right, that way is better